basiclibrary/src/main/java/inheritance/Shop.java

```diff
@@ -0,0 +1,82 @@
+package inheritance;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+public class Shop {
+    private String name;
+    private String description;
+    private int dollarSigns;
+    private int totalStars;
+    private LinkedList reviewList = new LinkedList();
+
+    public Shop(String name, String description, int dollarSigns){
+        this.name = name;
+        this.description = description;
+        this.dollarSigns = dollarSigns;
+        this.totalStars = 0;
+    }
+
+    public String getName(){
+        return this.name;
+    }
+
+    public String getDescription(){
+        return this.description;
+    }
+
+    public int getDollarSigns(){
+        return this.dollarSigns;
+    }
+
+    public void addReview(Review r){
+        if(this.reviewList == null){
+            this.reviewList.add(r);
+        }
+        else if(!checkList(r.toString() ) ){
+            this.reviewList.add(r);
+            this.totalStars += r.stars;
+        }
+        else{
+            throw new IllegalArgumentException("review already exists!");
+        }
+    }
+
+    public boolean checkList(String review){
+        if(this.reviewList == null){
+            return false;
+        }
+        else if (this.reviewList.size() == 1){
+            //System.out.println(this.reviewList.getFirst());
+            String tmp = this.reviewList.getFirst().toString();
+            if(tmp.equals(review) ){
+                return true;
+            }
+            else{
+                return false;
+            }
+        }
+        else{
+            Iterator it = this.reviewList.iterator();
```

Comment:
This definitely works, but an ArrayList can also be iterated over using a regular loop or an enhanced for loop, no need to create an iterator if you don't want to